java/vector/src/main/java/org/apache/arrow/vector/BitVectorHelper.java

```diff
@@ -329,7 +329,8 @@ public static ArrowBuf loadValidityBuffer(final ArrowFieldNode fieldNode,
     final int valueCount = fieldNode.getLength();
     ArrowBuf newBuffer = null;
     /* either all NULLs or all non-NULLs */
-    if (fieldNode.getNullCount() == 0 || fieldNode.getNullCount() == valueCount) {
+    if (sourceValidityBuffer == null && (fieldNode.getNullCount() == 0 ||
```

Comment:
the sourceValidity buffer will can only be null if getNullCount == 0.  It isn't clear to me why this code chose to always allocate a new buffer in this case as opposed to reusing the available one(I would have to trace it).

Reply:
sorry to be clearer.  The spec only allows for the bitmap to be null if all values are present, not if all values are null.

Reply:
Thanks @emkornfield , yea I missed that. Let me fix it.

Reply:
It turned out this is because the ORC adapter wraps null validity buffer from native side with `OrcMemoryJniWrapper`, which then gets converted into `ArrowBuf` with `memoryAddress` equals to 0 (i.e., it wraps a null buffer).

With the change in this PR, the Java side will think the `ArrowBuf` is valid and thus will skip allocating new buffer. However it is actually invalid and thus will cause a series of issues.

I added an additional check of the `ArrowBuf`'s memory address, to make sure it is indeed a null validity buffer.

Reply:
OK, found another issue with IPC between C++ and Java. In this case even though the source validity buffer is null from the C++ side, when deserializing it from the IPC `RecordBatch`, it is just a slice derived from the original message body (the IPC format seems to require validity buffer to always be present, even if all values are not-null in the Arrow array). Therefore, it won't be null or having a null memory address. 

To fix, I changed the check to be:

```
sourceValidityBuffer == null || sourceValidityBuffer.capacity() == 0;
```

The latter indicates that the buffer is empty.